Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ContestsController.cs

```diff
@@ -292,6 +326,54 @@ public ActionResult StartAsLab(LabStartInputModel inputModel)
             return new EmptyResult();
         }
 
+        [HttpGet]
+        public ActionResult ChangeTime(int contestId)
+        {
+            if (!this.CheckIfUserHasContestPermissions(contestId))
+            {
+                this.TempData.AddDangerMessage(GlobalConstants.NoPrivilegesMessage);
+                return this.RedirectToAction<ContestsController>(c => c.Index());
+            }
+
+            var contest = this.contestsData
+                .GetByIdQuery(contestId)
+                .Select(ChangeTimeForParticipantsViewModel.FromContest)
+                .FirstOrDefault();
+
+            if (contest != null)
+            {
+                return this.View(contest);
+            }
+
+            this.TempData.AddDangerMessage(Resource.Contest_not_valid);
+            return this.RedirectToAction<ContestsController>(c => c.Index());
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public ActionResult ChangeTime(ChangeTimeForParticipantsViewModel model)
+        {
+            if (!this.CheckIfUserHasContestPermissions(model.ContesId))
+            {
+                this.TempData.AddDangerMessage(GlobalConstants.NoPrivilegesMessage);
+                return this.RedirectToAction<ContestsController>(c => c.Index());
+            }
+
+            if (!this.contestsData.GetAllActive().Any(c => c.Id == model.ContesId))
+            {
+                this.TempData.AddDangerMessage(Resource.Contest_not_valid);
+                return this.RedirectToAction<ContestsController>(c => c.Index());
+            }
+
+            this.participantsData.ChangeTimeForActiveByContestIdAndMinutes(model.ContesId, model.TimeInMinutes);
+
+            this.TempData.AddInfoMessage(model.TimeInMinutes >= 0
+                ? string.Format(Resource.Added_time_to_participants_online, model.TimeInMinutes, model.ContestName)
+                : string.Format(Resource.Subtracted_time_from_participants_online, model.TimeInMinutes, model.ContestName));
```

Comment:
`model.TimeInMinutes` is a negative integer here. This results in a message that looks like '...subtracted -10 minutes...'